app/models/spree/user.rb

```diff
@@ -47,6 +48,25 @@ def self.admin_created?
       User.admin.count > 0
     end
 
+    def self.from_omniauth(auth)
+      where(auth.slice(:provider, :uid)).first_or_create do |user|
+        user.provider = auth.provider
+        user.uid = auth.uid
+        user.username = auth.info.nickname
```

Comment:
Will this `nickname` attribute always be present here? Or is there a chance it could be `nil`?

Also; are there any other attributes that could come from the `auth` object, like email? We'll need a good amount of tests here covering different cases and seeing how this interacts with validations on the user model (like if the user's email is nil)...

Reply:
> Will this nickname attribute always be present here?

That's a good catch, I don't think we'll even end up using that attribute at all. That line must have been suggested by Copilot ;-) 

> We'll need a good amount of tests here covering different cases 

👍 💯